src/tiffcomposite_int.cpp

```diff
@@ -270,24 +271,24 @@ void TiffDataEntry::setStrips(const Value* pSize, const byte* pData, size_t size
 #endif
     return;
   }
-  uint32_t size = 0;
+  size_t size = 0;
   for (size_t i = 0; i < pSize->count(); ++i) {
-    size += pSize->toUint32(i);
+    size = Safe::add<size_t>(size, pSize->toUint32(i));
   }
-  auto offset = pValue()->toUint32(0);
-  // Todo: Remove limitation of JPEG writer: strips must be contiguous
-  // Until then we check: last offset + last size - first offset == size?
-  if (pValue()->toUint32(pValue()->count() - 1) + pSize->toUint32(pSize->count() - 1) - offset != size) {
+  const size_t offset = pValue()->toUint32(0);
+  if (size > sizeData || offset > sizeData - size || baseOffset > sizeData - size - offset) {
 #ifndef SUPPRESS_WARNINGS
     EXV_WARNING << "Directory " << groupName(group()) << ", entry 0x" << std::setw(4) << std::setfill('0') << std::hex
-                << tag() << ": Data area is not contiguous, ignoring it.\n";
+                << tag() << ": Data area exceeds data buffer, ignoring it.\n";
 #endif
     return;
   }
-  if (offset > sizeData || size > sizeData || baseOffset + offset > sizeData - size) {
+  // Todo: Remove limitation of JPEG writer: strips must be contiguous
```

Comment:
[suggestion] I noticed over time that we have many different ways to mark `TODO` comments. Some IDEs let you use different kind of `TODO` markers but others are less customizable. 

I would like to propose to use always the same keyword for these comments. The 2 options I like more are `TODO` or use the doxygen style `/// \todo`.

Reply:
This is an old TODO that I only moved, so I'll leave it like this in this PR. But I'll do a separate PR to standardize all the TODOs